tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -137,9 +137,13 @@ internal TracerManager CreateTracerManager(
             agentWriter ??= GetAgentWriter(settings, settings.TracerMetricsEnabled ? statsd : null, rates => sampler.SetDefaultSampleRates(rates), discoveryService);
             scopeManager ??= new AsyncLocalScopeManager();
 
-            if (runtimeMetricsEnabled)
+            if (runtimeMetricsEnabled && runtimeMetrics is { })
```

Comment:
maybe to avoid double testing `runtimeMetricsEnabled` 

```
if (!runtimeMetricsEnabled)
{
    runtimeMetrics = null;
}
else if (runtimeMetrics is { })
{
    runtimeMetrics.UpdateStatsd(statsd);
}
else
{
    runtimeMetrics = new RuntimeMetricsWriter(
        statsd,
        TimeSpan.FromSeconds(10),
        settings.IsRunningInAzureAppService);
}
````